Kernel/Devices/Storage/NVMe/NVMeQueue.h

```diff
@@ -56,9 +58,27 @@ class NVMeQueue : public AtomicRefCounted<NVMeQueue> {
 
 protected:
     u32 process_cq();
+
+    // Updates the shadow buffer and returns if mmio is needed
+    bool update_shadow_buf(u16 new_value, u32* dbbuf, u32* ei)
+    {
+        bool need_mmio = true;
+        u16 old;
```

Comment:
You don't need to declare your variables at the top - just define them where you use them. `need_mmio` does not need to be initialized to anything that way.

Reply:
Absolutely we can ;) . I will push a new update,